src/js/components/Drop/DropContainer.js

```diff
@@ -209,23 +209,61 @@ const DropContainer = forwardRef(
             less room in the other direction leave the Drop in its current
             position. */
           if (
+            /* The bottom of the drop is aligned to the top of the target 
+            and the top of the drop is either at the top edge of the viewport
+            or beyond it. If there is room to display the drop below the target
+            change the drop's position so that the top of the drop is aligned 
+            to the bottom of the target. */
             responsive &&
```

Comment:
The top few conditions check for `responsive &&` then the bottom ones don't. Would it be better to write this logic as:
```
if (responsive) {
   if (
      align.bottom === 'top' && 
      // drop is overflowing above window
      targetRect.top - containerRect.height <= 0 && 
      // there's space below the target 
      targetRect.bottom + containerRect.height < windowHeight
   ) { 
       ... 
   } else if () {
   } ...
} else if (align.top === 'top') {...}
...
```
just would save a few condition checks for the non-responsive cases

Reply:
If we restructure this then we run into issues where if responsive is true but it doesn't meet any of those top 4 conditions then it will never reach the other else if statements or the else statement at the end.

Reply:
Ah thank you for clarifying.